integrations/snyk/snyk/client.py

```diff
@@ -0,0 +1,268 @@
+        if "projects" in event.attributes:
+            all_projects = event.attributes["projects"]
+            projects_to_yield = (
+                [
+                    project
+                    for project in all_projects
+                    if project.get("__target", {}).get("data", {}).get("id")
+                    == target_id
+                ]
+                if target_id
+                else all_projects
+            )
+            yield projects_to_yield
+            return
+
+        url = f"/orgs/{self.org_id}/projects"
+        query_params = {
+            "version": self.snyk_api_version,
+            "meta.latest_issue_counts": "true",
+            "expand": "target",
+        }
+
+        async for projects in self._get_paginated_resources(
+            base_url=self.rest_api_url, url_path=url, query_params=query_params
+        ):
+            all_projects = []
+
+            for project in projects:
+                enriched_project = await self.enrich_project(project)
+                all_projects.append(enriched_project)
+
+            event.attributes.setdefault("projects", []).extend(all_projects)
+
+            projects_to_yield = (
+                [
+                    project
+                    for project in all_projects
+                    if project.get("__target", {}).get("data", {}).get("id")
+                    == target_id
+                ]
+                if target_id
+                else all_projects
+            )
+            yield projects_to_yield
+
+    async def get_single_target_by_project_id(self, project_id: str) -> dict[str, Any]:
+        project = await self.get_single_project({"id": project_id})
+        target_id = project.get("__target", {}).get("data", {}).get("id")
+
+        url = f"{self.rest_api_url}/orgs/{self.org_id}/targets/{target_id}"
+        query_params = {"version": f"{self.snyk_api_version}~beta"}
+
+        response = await self._send_api_request(
+            url=url, method="GET", query_params=query_params
+        )
+
+        target = response.get("data", {})
+        async for projects_data_of_target in self.get_paginated_projects(
+            target.get("id")
+        ):
+            target.setdefault("__projects", []).extend(projects_data_of_target)
+        return target
+
+    async def get_paginated_targets(
+        self,
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        url = f"/orgs/{self.org_id}/targets"
+        query_params = {"version": f"{self.snyk_api_version}~beta"}
+        async for targets in self._get_paginated_resources(
+            base_url=self.rest_api_url, url_path=url, query_params=query_params
+        ):
+            targets_with_project_data = []
+            for target_data in targets:
+                async for projects_data_of_target in self.get_paginated_projects(
+                    target_data.get("id")
+                ):
+                    target_data.setdefault("__projects", []).extend(
```

Comment:
is it possible that the target already have __projects?

Reply:
Yeap, because I am doing it in generator inside a generator